polaris-core/src/main/java/org/apache/polaris/core/config/BehaviorChangeConfigurationSince.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.polaris.core.config;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * An annotation that should be applied to all BehaviorChangeConfiguration instances. The value here
+ * is the Polaris version when the flag was introduced. A spotless check should validate that
+ * behavior change flags are not present for more than 2 minor releases or 1 major release
+ */
+@Target(ElementType.FIELD)
+@Retention(RetentionPolicy.RUNTIME)
+public @interface BehaviorChangeConfigurationSince {
```

Comment:
Side note: it would be great to consider this annotation in `tools/config-docs/generator`

Reply:
Do you mean to move it there?

Reply:
No, I mean auto-generated config docs should ideally consider this annotation

Reply:
I see, I've looked through the `generator` package but I'm not sure I fully understand what it would mean for the docs to consider the annotation. I guess the idea would be to have the docs about each BehaviorChangeConfiguration describe what version the configuration was created in, etc.? In fact, I don't know if we even want to document the `BehaviorChangeConfiguration`s

Reply:
I believe, in an open source project all config that a user may want to change should be documented... but we can leave `BehaviorChangeConfigurations` for later, of course.